### PR TITLE
redesign handler workflow and add interface

### LINKED ISSUE #22: Add Interface Message Handler
Add a interface for message handlers

### PR DESCRIPTION
**Pull request by jens1o**
Classes are now instantiated on the fly. Thus, the `canHandle` method
needed to be made static.

### Changelog
- Created messagehandler interface
- instantiate them on the fly

resolves #22 #27

### CODE REVIEW

**Comment by jens1o** [comment]
(will update the tests if you're okay with the general approach)

**Comment by micschwarz** [comment]
1. Fix pull request description
2. Fix tests

File: src/message/handler/JQueryMessageHandler.ts [hunk 1]
```diff
@@ -1,13 +1,16 @@
...
+import { Message, Client } from 'discord.js';
+import { jQueryEmoticon } from '../../constants';
 
 export default class JQueryMessageHandler extends AbstractMessageHandler {
-    constructor() {
-        super(/jquery/gim);
+
+    private static readonly MESSAGE_REGEX: RegExp = /jquery/gim;
+
+    static canHandle(message: string): boolean {
```
Comment (micschwarz) [inline src/message/handler/JQueryMessageHandler.ts]: Function can be put in abstract class -> less duplicate code

Comment (jens1o) [reply src/message/handler/JQueryMessageHandler.ts]: No. You cannot force a class to have static functions and variables.

**Pull request merged**